### PR TITLE
#961 - upgraded to qulice 0.15.2

### PR DESCRIPTION
**Pull request by danff**
Fix for #961 - qulice upgrade
- Added qulice ver 0.15.2 to pom.xml files;
- Fixed some vriable names violations;
- Added puzzle to pom.xml files with ecludes for classes vilating checkstyle/pmd checks;

### CODE REVIEW

**Comment by karato** [comment]
@danff Thanks, someone will review your pull request soon

**Comment by karato** [comment]
@krzyk do this review

File: netbout-client/pom.xml [hunk 1]
```diff
@@ -189,6 +190,27 @@
                                  Findbugs filters in current version of Qulice
                                 -->
                                 <exclude>findbugs:~com.netbout.client.*</exclude>
+                                <!--
+                                 puzzle #961:30min declared in netbout-web/pom.xml.
```
Comment (krzyk) [inline netbout-client/pom.xml]: @danff what is this comment for? It is not a correct puzzle

File: netbout-client/pom.xml [hunk 2]
```diff
@@ -178,6 +178,7 @@
                     <plugin>
                         <groupId>com.qulice</groupId>
                         <artifactId>qulice-maven-plugin</artifactId>
+                        <version>0.15.2</version>
```
Comment (krzyk) [inline netbout-client/pom.xml]: @danff you should remove this and add pluginManagement section in netbout/pom.xml with this qulice version

File: netbout-spi/pom.xml [hunk 3]
```diff
@@ -88,6 +88,7 @@
                     <plugin>
                         <groupId>com.qulice</groupId>
                         <artifactId>qulice-maven-plugin</artifactId>
+                        <version>0.15.2</version>
```
Comment (krzyk) [inline netbout-spi/pom.xml]: @danff remove this

File: netbout-web/pom.xml [hunk 4]
```diff
@@ -726,6 +726,7 @@
                     <plugin>
                         <groupId>com.qulice</groupId>
                         <artifactId>qulice-maven-plugin</artifactId>
+                        <version>0.15.2</version>
```
Comment (krzyk) [inline netbout-web/pom.xml]: @danff remove this

File: netbout-spi/src/main/java/com/netbout/spi/Searchable.java [hunk 5]
```diff
@@ -35,6 +35,7 @@
  * @author Ivan Inozemtsev (ivan.inozemtsev@gmail.com)
  * @version $Id$
  * @since 2.15.2
+ * @param <T> type of the items
```
Comment (krzyk) [inline netbout-spi/src/main/java/com/netbout/spi/Searchable.java]: @danff should start from uppercase letter

File: netbout-web/src/main/java/com/netbout/Launch.java [hunk 6]
```diff
@@ -43,6 +43,7 @@
  * @author Yegor Bugayenko (yegor@teamed.io)
  * @version $Id$
  * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ * @since 1.0
```
Comment (krzyk) [inline netbout-web/src/main/java/com/netbout/Launch.java]: @danff since version is wrong, correct one is 2.14

File: netbout-web/pom.xml [hunk 7]
```diff
@@ -743,6 +744,40 @@
...
+	                            <!--
+                                @todo #961:30min These excludes were added
+                                 during upgrade to qulice 0.15.2. New version of
+                                 qulice adds new violations that need to be fixed.
+                                 There are also excludes added to netbout-client/pom.xml.
+                                 They are also part of this puzzle and need to be
+                                 fixed.
+                             	-->
+                             	<exclude>checkstyle:/src/main/java/com/netbout/cached/CdInbox.java</exclude>
+								<exclude>checkstyle:/src/main/java/com/netbout/cached/CdMessages.java</exclude>
```
Comment (krzyk) [inline netbout-web/pom.xml]: @danff why the indetnation is off here and in following lines?

File: netbout-web/pom.xml [hunk 8]
```diff
@@ -743,6 +744,40 @@
...
+								<exclude>checkstyle:/src/test/java/com/netbout/dynamo/DyInboxITCase.java</exclude>
+								<exclude>checkstyle:/src/test/java/com/netbout/dynamo/DyMessagesITCase.java</exclude>
+                             	<exclude>checkstyle:/src/test/java/com/netbout/dynamo/DyUserITCase.java</exclude>
+                             	<exclude>checkstyle:/src/test/java/com/netbout/email/EmAliasITCase.java</exclude>
+                             	<exclude>checkstyle:/src/test/java/com/netbout/email/EmMessagesITCase.java</exclude>
+                             	<exclude>checkstyle:/src/test/java/com/netbout/rest/bout/TkBoutITCase.java</exclude>
+                             	<exclude>checkstyle:/src/test/java/com/netbout/rest/bout/TkDeleteITCase.java</exclude>
+                             	<exclude>checkstyle:/src/test/java/com/netbout/rest/bout/XeMessageTest.java</exclude>
+                             	<exclude>checkstyle:/src/test/java/com/netbout/rest/TkAppITCase.java</exclude>
+								<exclude>pmd:/src/main/java/com/netbout/dynamo/DyBase.java</exclude>
```
Comment (krzyk) [inline netbout-web/pom.xml]: @danff here again, indentation is wrong

File: netbout-web/src/main/java/com/netbout/cached/CdPageable.java [hunk 9]
```diff
@@ -43,6 +43,7 @@
  * @author Yegor Bugayenko (yegor@teamed.io)
  * @version $Id$
  * @since 2.10.3
+ * @param <T> items type
```
Comment (krzyk) [inline netbout-web/src/main/java/com/netbout/cached/CdPageable.java]: @danff should start from uppercase letter

File: netbout-spi/src/main/java/com/netbout/spi/Pageable.java [hunk 10]
```diff
@@ -35,6 +35,7 @@
  * @author Yegor Bugayenko (yegor@teamed.io)
  * @version $Id$
  * @since 2.0
+ * @param <T> type of the items
```
Comment (krzyk) [inline netbout-spi/src/main/java/com/netbout/spi/Pageable.java]: @danff should start from uppercase letter

File: netbout-web/src/main/java/com/netbout/email/GmailViewAction.java [hunk 11]
```diff
@@ -34,6 +34,7 @@
  * Integrates with Gmail's View Actions.
  * @author Matteo Barbieri (barbieri.matteo@gmail.com)
  * @version $Id$
+ * @since 2.0
```
Comment (krzyk) [inline netbout-web/src/main/java/com/netbout/email/GmailViewAction.java]: @danff since 2.15

File: netbout-client/pom.xml (hunk 1 above)
Comment (danff) [inline netbout-client/pom.xml]: And it should not be. The puzzle is declared in netbout-web/pom.xml and declaring here also will raise puzzle duplication error in travis.

**Comment by danff** [comment]
@krzyk Issues addressed. thanks. The only question is how to create a puzzle in 2 different files? In our case we have excludes in netbout-client/pom.xml and also in netbout-web/pom.xml as part of the same puzzle.

**Comment by krzyk** [comment]
@danff you create just one puzzle, and mention where the changes have to be made in that puzzle, so remove the comment in the other pom file

File: netbout-web/src/test/java/com/netbout/rest/TkInboxTest.java [hunk 12]
```diff
@@ -94,14 +94,14 @@ public void searchesBouts() throws Exception {
...
         final Inbox inbox = aliases.iterate().iterator().next().inbox();
-        final Bout firstBout = inbox.bout(inbox.start());
-        final String firstTitle = "bout1 title";
-        firstBout.rename(firstTitle);
-        final Bout secondBout = inbox.bout(inbox.start());
-        final String secondTitle = "bout2 title";
-        secondBout.rename(secondTitle);
-        firstBout.messages().post("hello");
-        secondBout.messages().post("world");
+        final Bout firstbout = inbox.bout(inbox.start());
```
Comment (krzyk) [inline netbout-web/src/test/java/com/netbout/rest/TkInboxTest.java]: @danff you could name it just `first`

File: netbout-web/src/test/java/com/netbout/rest/TkInboxTest.java [hunk 13]
```diff
@@ -94,14 +94,14 @@ public void searchesBouts() throws Exception {
...
+        final String firsttitle = "bout1 title";
+        firstbout.rename(firsttitle);
+        final Bout secbout = inbox.bout(inbox.start());
```
Comment (krzyk) [inline netbout-web/src/test/java/com/netbout/rest/TkInboxTest.java]: @danff this one would be `second`

File: netbout-client/pom.xml [hunk 14]
```diff
@@ -189,6 +189,28 @@
                                  Findbugs filters in current version of Qulice
                                 -->
                                 <exclude>findbugs:~com.netbout.client.*</exclude>
+                                <!--
+                                 Additional to puzzle #961:30min declared in
```
Comment (krzyk) [inline netbout-client/pom.xml]: @danff this is to be removed, and appropriate comment added to the other puzzle (just mention that in netbout-client/pom.xml and checkstyle and pmd excludes should bre removed), you could also create two puzzles, if you think that the work in netbout-client is for 30mins and in netbout-web is also for 30mins.
You just have to make the text different.

**Comment by krzyk** [comment]
@danff please see my additional comments

File: netbout-web/src/test/java/com/netbout/rest/TkInboxTest.java (hunk 12 above)
Comment (danff) [inline netbout-web/src/test/java/com/netbout/rest/TkInboxTest.java]: actually there are 4 variables, firstBout and firstTitle, secondBout and secondTitle.

File: netbout-web/src/test/java/com/netbout/rest/TkInboxTest.java (hunk 13 above)
Comment (danff) [inline netbout-web/src/test/java/com/netbout/rest/TkInboxTest.java]: actually there are 2 variables, secondBout and secondTitle.

File: netbout-web/src/test/java/com/netbout/rest/TkInboxTest.java (hunk 12 above)
Comment (krzyk) [inline netbout-web/src/test/java/com/netbout/rest/TkInboxTest.java]: @danff please always add who you are addressing you response to, QAs don't like unaddressed ones.
As for the variables, I'm writing about the Bout one, you can't do anything about title, but you can about this one.

File: netbout-web/src/test/java/com/netbout/rest/TkInboxTest.java (hunk 13 above)
Comment (krzyk) [inline netbout-web/src/test/java/com/netbout/rest/TkInboxTest.java]: @danff add `@krzyk`, and I'm writting bouot the Bout here also

**Comment by danff** [comment]
@krzyk updated comments in xmls and renamed the variables. Please check if it is ok now.

File: netbout-web/src/test/java/com/netbout/rest/TkInboxTest.java [hunk 15]
```diff
@@ -94,14 +94,14 @@ public void searchesBouts() throws Exception {
...
         final Inbox inbox = aliases.iterate().iterator().next().inbox();
-        final Bout firstBout = inbox.bout(inbox.start());
-        final String firstTitle = "bout1 title";
-        firstBout.rename(firstTitle);
-        final Bout secondBout = inbox.bout(inbox.start());
-        final String secondTitle = "bout2 title";
-        secondBout.rename(secondTitle);
-        firstBout.messages().post("hello");
-        secondBout.messages().post("world");
+        final Bout fbout = inbox.bout(inbox.start());
```
Comment (krzyk) [inline netbout-web/src/test/java/com/netbout/rest/TkInboxTest.java]: @danff the idea was to get rid of compound names, you can do that with at least two variables here

**Comment by krzyk** [comment]
@danff almost :) see my comment

**Comment by danff** [comment]
@krzyk new commit added

**Comment by krzyk** [comment]
@danff why are you keep changing the variables for title? they were OK, (because we can't have them `first` and `second` also), please bring them back

**Comment by danff** [comment]
@krzyk titles are back

**Comment by danff** [comment]
@krzyk closed by mistake

**Comment by krzyk** [comment]
@danff thanks

**Comment by krzyk** [comment]
@rultor merge pls

**Comment by dmzaytsev** [comment]
@rultor merge

**Comment by karato** [comment]
@ypshenychka review this ticket please, for compliance with our [quality rules](http://at.teamed.io/qa.html)

**Comment by karato** [comment]
@rultor please deploy

**Comment by ypshenychka** [comment]
@danff 
According to our [QA Rules](http://at.teamed.io/qa.html):
>Messages in a ticket always start with a name of a user they are addressed to.

Please correct your code review comments by indicating an addressee in the beginning.

**Comment by danff** [comment]
@ypshenychka Thanks, fixed.

**Comment by ypshenychka** [comment]
@danff Thank you.

**Comment by ypshenychka** [comment]
@karato Quality is acceptable.

**Comment by karato** [comment]
> @karato Quality is acceptable.

@ypshenychka thanks for the review, we'll try better next time

**Comment by karato** [comment]
@krzyk I added 10 mins to @ypshenychka (for QA review) in transaction 77327829. paid, thanks, added **38 mins** to your account, payment `77327849`, 24 hours and 9 mins was spent. review comments (c=23) added as a bonus. +38 added to your rating, current score is: [+5849](http://www.netbout.com/b/35035?open=rating)

**Pull request merged**